client/configuration_variable_test.go

```diff
@@ -0,0 +1,179 @@
+package client_test
+
+import (
+	. "github.com/env0/terraform-provider-env0/client"
+	"github.com/golang/mock/gomock"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/ginkgo/extensions/table"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("Configuration Variable", func() {
+	mockConfigurationVariable := ConfigurationVariable{
+		Id:             "config-var-id-789",
+		Name:           "configName",
+		Value:          "configValue",
+		OrganizationId: organizationId,
+		IsSensitive:    true,
+		Scope:          "PROJECT",
+		Type:           0,
+		ScopeId:        "project-123",
+		UserId:         "user|123",
+	}
+
+	Describe("ConfigurationVariableCreate", func() {
+		var createdConfigurationVariable ConfigurationVariable
+
+		BeforeEach(func() {
+			mockOrganizationIdCall(organizationId)
+
+			expectedCreateRequest := []map[string]interface{}{{
+				"name":           mockConfigurationVariable.Name,
+				"isSensitive":    mockConfigurationVariable.IsSensitive,
+				"value":          mockConfigurationVariable.Value,
+				"organizationId": organizationId,
+				"scopeId":        mockConfigurationVariable.ScopeId,
+				"scope":          Scope(mockConfigurationVariable.Scope),
+				"type":           ConfigurationVariableType(mockConfigurationVariable.Type),
+			}}
+
+			httpCall = mockHttpClient.EXPECT().
+				Post("configuration", expectedCreateRequest, gomock.Any()).
+				Do(func(path string, request interface{}, response *[]ConfigurationVariable) {
+					*response = []ConfigurationVariable{mockConfigurationVariable}
+				})
+
+			createdConfigurationVariable, _ = apiClient.ConfigurationVariableCreate(
+				mockConfigurationVariable.Name,
+				mockConfigurationVariable.Value,
+				mockConfigurationVariable.IsSensitive,
+				Scope(mockConfigurationVariable.Scope),
+				mockConfigurationVariable.ScopeId,
+				ConfigurationVariableType(mockConfigurationVariable.Type),
+				nil,
+			)
+		})
+
+		It("Should get organization id", func() {
+			organizationIdCall.Times(1)
+		})
+
+		It("Should send POST request with params", func() {
+			httpCall.Times(1)
+		})
+
+		It("Should return created configuration variable", func() {
+			Expect(createdConfigurationVariable).To(Equal(mockConfigurationVariable))
+		})
+	})
+
+	Describe("ConfigurationVariableDelete", func() {
+		BeforeEach(func() {
+			httpCall = mockHttpClient.EXPECT().Delete("configuration/" + mockConfigurationVariable.Id)
+			apiClient.ConfigurationVariableDelete(mockConfigurationVariable.Id)
+		})
+
+		It("Should send DELETE request with project id", func() {
+			httpCall.Times(1)
+		})
+	})
+
+	Describe("ConfigurationVariableUpdate", func() {
+		var updatedConfigurationVariable ConfigurationVariable
+
+		BeforeEach(func() {
+			mockOrganizationIdCall(organizationId)
+
+			newName := "new-" + mockConfigurationVariable.Name
+			newValue := "new-" + mockConfigurationVariable.Value
+
+			expectedUpdateRequest := []map[string]interface{}{{
```

Comment:
there are a common fields with `expectedCreateRequest` maybe we can dry by implement common expectedRequest

Reply:
I couldn't find a nice way to copy/clone the map object in Go ...
So leaving as is for now ...